webapp/public/run.html

```diff
@@ -168,8 +169,14 @@
                     window.parent.postMessage(msg.data, "*");
                 }
                 else if (mpRole && data.type === "multiplayer") {
-                    // propagate message packets to parent frames
-                    window.parent.postMessage(msg.data, "*");
+                    // Multiplayer messages are bi-directional
+                    if (mpRole === msg.data.origin) {
```

Comment:
would a

```
if (event.source !== window.parent)
```

work to prevent passing back / forth? I believe we do that elsewhere so it feels like it must have just gotten missed / dropped here